### PR TITLE
Clean up comment sediment across maintained source and configuration

### PR DESCRIPTION
**Pull request by arthur-debert**
## Summary

Repository-wide comments-only cleanup delivered in three ordered delegated phases on one PR:

1. Production Rust comments and doc comments under `crates/`, excluding test-only regions.
2. Rust test comments across dedicated test/support files and inline `#[cfg(test)]` modules.
3. Hand-owned YAML/TOML comments; eligible Markdown was inspected but contained no HTML comments.

The cleanup removes process chronology, review/workstream provenance, duplicated explanations, stale future-tense claims, and self-evident narration. It preserves design rationale, compatibility and wire-format contracts, security boundaries, platform/tool quirks, edge-case intent, and durable ADR/spec/issue references.

## Verification

- `pixi run test` — 1,302/1,302 passed.
- `pixi run lint` — 13/13 passed; commit and push hooks passed.
- `cargo doc -p dodot-lib --no-deps` completed; no new documentation failure.
- Final cumulative diff audit over `origin/main..HEAD`: 125 files, all changed spans are recognized comments/doc comments, with seven trailing-comment edits whose executable prefixes are byte-identical; zero blank-line deltas.
- Shipit-managed files, projections, `.shipit.toml` machine state, and every marked `pixi.toml` managed block are byte-identical.
- No Rust code, test logic, identifiers, signatures, strings, YAML/TOML data, ordinary Markdown prose, generated files, or legal text changed.

## Context

**Why this approach.** Each large phase was fanned out into non-overlapping scopes, integrated in sequence on `issues/253/work`, and independently audited before the next phase. A final read-only audit reviewed all 1,112 hunks and its findings were corrected in `dd7f136`.

**Ownership boundary.** Workflow callers and consumer declaration comments are hand-owned even when they route through Shipit. Shipit-managed blocks and projected agent/workflow support files were excluded. Markdown product docs were inspected without modification because ordinary prose is not a comment and no eligible HTML comments existed.

**Out of scope.** This PR does not rewrite stale `.lex` reference paths, resolve unrelated terminology drift, fill documentation gaps, or change any executable behavior.

closes #253

### CODE REVIEW

File: crates/dodot-lib/src/commands/template_install_filter.rs [hunk 1]
```diff
@@ -28,8 +28,8 @@ use crate::packs::orchestration::ExecutionContext;
 use crate::{DodotError, Result};
 
 /// The `.git/config` snippet this command writes. Public so
-/// `dodot template show-filter` (future) can print it for users
-/// who want to install by hand.
+/// `dodot template show-filter` can print it for users who want to
+/// install by hand.
```
[the remark this answers is not in the document]
Comment (arthur-debert) [reply crates/dodot-lib/src/commands/template_install_filter.rs]: Fixed in c0039de: the comment now attributes the inspection path to the post-up prompt and no longer names a nonexistent subcommand. The adjacent show_filter documentation already describes that prompt branch accurately.

File: crates/dodot-lib/src/secret/secret_string.rs [hunk 2]
```diff
@@ -39,9 +39,9 @@ use zeroize::Zeroize;
 /// A resolved secret value, held briefly in process memory.
 ///
 /// Construct via [`SecretString::new`]; read via [`SecretString::expose`].
-/// Zeroes its buffer on drop. Has no `Debug` / `Display` / `Serialize`
-/// implementations; printing one through any of those paths produces
-/// `<redacted>`.
+/// Zeroes its buffer on drop. `Debug` prints `<redacted>`; `Display` and
+/// `Serialize` are not implemented at all. See the module docs for what
+/// this wrapper does and does not protect against.
```
[the remark this answers is not in the document]
Comment (arthur-debert) [reply crates/dodot-lib/src/secret/secret_string.rs]: Fixed in c0039de: the struct docs now name the exact SecretString(<redacted>, len=N) Debug representation. I also checked the module-level contract, which already matches the implementation.

File: crates/dodot-lib/src/datastore/mod.rs [hunk 3]
```diff
@@ -16,11 +16,10 @@ use crate::Result;
...
 ///
-/// Mirrors the spec in #169: run-once handlers consult this to decide
-/// between *first-time-run* (NeverRan → execute), *already up to
-/// date* (RanCurrent → skip silently), and *file edited since last
-/// run* (RanDifferent → skip with notice, user runs `--force` to
-/// apply).
+/// Run-once handlers consult this to decide between
+/// *first-time-run* (NeverRan → execute), *already up to date*
+/// (RanCurrent → skip silently), and *file edited since last run*
+/// (RanDifferent → skip with notice, user runs `--force` to apply).
```
[the remark this answers is not in the document]
Comment (arthur-debert) [reply crates/dodot-lib/src/datastore/mod.rs]: Fixed in c0039de: this now names --provision-rerun. I swept the cumulative diff for the same stale rerun-flag claim and corrected the matching module documentation in execution/run.rs as well.
